testsuite/tests/parsetree/source.ml

```diff
@@ -7443,3 +7443,9 @@ let g y =
   f ~(y:int)
 
 let goober a = match a with C (type a b) y -> y
+
+
+(** With constraints *)
+
+module type s = sig type ('a,'b) t end with type (-!'a, !+'b) t = 'b -> 'a list
+module type s = sig type ('a,'b) t end with type (!-'a, +!'b) t := 'b -> 'a list
```

Comment:
What gets printed in the case where the user writes just `with type ('a, 'b) t = ...`? Could you add this to the test?

Reply:
The user get back `with type ('a, 'b) t = ` in this case.